Mathlib/Tactic/Cases.lean

```diff
@@ -63,7 +63,7 @@ elab (name := induction') "induction' " tgts:(casesTarget,+)
     withArg:((" with " (colGt binderIdent)+)?)
     genArg:((" generalizing " (colGt ident)+)?) : tactic => do
   let targets ← elabCasesTargets tgts.1.getSepArgs
-  let g ← getMainGoal
+  let g :: gs ← getUnsolvedGoals | throwNoGoalsToBeSolved
```

Comment:
Does that mean that every tactic using `withMainContext` is broken because it might run on already solved goals?

Reply:
No, `withMainContext` calls `getMainGoal` which selects the first *unsolved* goal (and also updates the goal list to remove the solved goals, which might be surprising for a `getFoo` function but generally results in tactics doing the right thing wrt subsequent goal list modifications).